cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -355,30 +369,85 @@ public KafkaAssemblyOperator(Vertx vertx, PlatformFeaturesAvailability pfa,
+     *
+     * @param reconciliationState   Reconciliation State
+     *
+     * @return                      Future with Reconciliation State
+     */
+    Future<ReconciliationState> reconcileKafkaExporter(ReconciliationState reconciliationState)    {
+        return reconciliationState.getKafkaExporterDescription()
                 .compose(state -> state.kafkaExporterServiceAccount())
                 .compose(state -> state.kafkaExporterSecret(this::dateSupplier))
                 .compose(state -> state.kafkaExporterDeployment())
-                .compose(state -> state.kafkaExporterReady())
+                .compose(state -> state.kafkaExporterReady());
+    }
 
-                .compose(state -> state.getJmxTransDescription())
+    /**
+     * Run the reconciliation pipeline for the JMX Trans cluster
+     *
+     * @param reconciliationState   Reconciliation State
+     *
+     * @return                      Future with Reconciliation State
+     */
+    Future<ReconciliationState> reconcileJmxTrans(ReconciliationState reconciliationState)    {
+        return reconciliationState.getJmxTransDescription()
                 .compose(state -> state.jmxTransServiceAccount())
                 .compose(state -> state.jmxTransConfigMap())
                 .compose(state -> state.jmxTransDeployment())
-                .compose(state -> state.jmxTransDeploymentReady())
+                .compose(state -> state.jmxTransDeploymentReady());
+    }
+
+    Future<Void> reconcile(ReconciliationState reconcileState)  {
+        Promise<Void> chainPromise = Promise.promise();
+
+        reconcileState.initialStatus()
+                // Preparation steps => prepare cluster descriptions, handle CA creation or changes
+                .compose(state -> state.reconcileCas(this::dateSupplier))
+                .compose(state -> state.clusterOperatorSecret(this::dateSupplier))
+                .compose(state -> state.getKafkaClusterDescription())
+                .compose(state -> state.getZookeeperDescription()) // Has to be before the rollingUpdateForNewCaKey
+                .compose(state -> state.prepareVersionChange())
+                // Roll everything if a new CA is added to the trust store.
+                .compose(state -> state.rollingUpdateForNewCaKey())
 
+                // Run ZooKeeper reconciliation
+                .compose(state -> reconcileZooKeeper(state))
+                // Run Kafka reconciliation
+                .compose(state -> reconcileKafka(state))
+                // Run EntityOperator reconciliation
+                .compose(state -> reconcileEntityOperator(state))
+                // Run Cruise Control reconciliation
+                .compose(state -> reconcileCruiseControl(state))
+                // Run Kafka Exporter reconciliation
+                .compose(state -> reconcileKafkaExporter(state))
+                // Run JMXTrans reconciliation
+                .compose(state -> reconcileJmxTrans(state))
```

Comment:
maybe the comments on each of the above lines are too much, I think you defined well self-descriptive method names? :-)

Reply:
Ok, I reduced the comments in this place.